fire/inspectutils.py

```diff
@@ -63,46 +77,40 @@ def GetArgSpec(fn):
   Args:
     fn: The function to analyze.
   Returns:
-    A named tuple of type inspect.ArgSpec with the following fields:
+    An object containing the following fields:
       args: A list of the argument names accepted by the function.
       varargs: The name of the *varargs argument or None if there isn't one.
-      keywords: The name of the **kwargs argument or None if there isn't one.
+      varkw: The name of the **kwargs argument or None if there isn't one.
       defaults: A tuple of the defaults for the arguments that accept defaults.
+      kwonlyargs: A list of argument names that must be passed with a keyword.
+      kwonlydefaults: A dictionary of keyword only arguments and their defaults.
+      annotations: A dictionary of arguments and their annotated types.
   """
   fn, skip_arg = _GetArgSpecFnInfo(fn)
 
   try:
 
     if six.PY2:
       argspec = inspect.getargspec(fn)
-      keywords = argspec.keywords
+      args = argspec.args[1:] if skip_arg else argspec.args
+      annotations = getattr(fn, '__annotations__', None)
+      fullspec = _FullArgSpec(args, *argspec[1:], annotations=annotations)
```

Comment:
It is probably too clever. I'll stop inling the if and just pop off the first argument. I'll probably change FullArgSpec so that it takes a function. Then passing hidden arguments won't be a problem. Since FullArgSpec no longer really behaves like inspect.FullArgSpec, it probably also needs a new name.
``